### DISCUSSION TITLE
VR headset support

### DISCUSSION THREAD

**Discussion by mercurial-moon** [airsdk/Adobe-Runtime-Support#3423] [General]
Is there a plan to support VR headsets like Oculus Quest.
I guess the headset is based on android / arm architecture so app running should not be an issue, it may not run in VR mode though.
However are there plans, that enable developers to build VR games / apps using AIR sdk's.

**Comment by Fancy2209** [comment]
there were ANE to allow this for the Oculus DK2, should be possible to do the same for the Quest SDKs?

**Comment by ajwfrost** [comment]
@mercurial-moon sorry, we had looked into this but I had neglected to update here .. 
It looks like this would mean updates in the Android development/platform configuration - so we would need to build using the [OpenXR Mobile SDK](https://developers.meta.com/horizon/documentation/native/android/mobile-intro) and it looks like the AR/VR stuff needs us to hook into the run loop ..

It might also be possible via an ANE I guess.

There is a bit of a question open around the platform/build requirements though.. we can perhaps try setting up a test app for this and then see how it might work with AIR. It may also be something that would be helped by some changes we're planning around the use of an AIR activity separately i.e. within another app - which are needed to properly support the Android lifecycle, but involve a lot of fundamental changes to how the runtime loads and is killed or could be reused..

So:
- short term - potentially an ANE solution may be viable
- long term - we can look at adding this support, probably in conjunction with a simplification/extension of how AIR can be used within other Android apps

thanks
